src/io/flutter/vmService/HeapMonitor.java

```diff
@@ -134,40 +128,48 @@ private void collectMemoryUsage() {
       return;
     }
 
-    // Stash count so we can know when we've processed them all.
-    final int isolateCount = isolateRefs.size();
-
     final List<MemoryUsage> memoryUsage = new ArrayList<>();
 
+    final CountDownLatch latch = new CountDownLatch(isolateRefs.size());
+
     for (IsolateRef isolateRef : isolateRefs) {
       vmServiceWrapper.getVmService().getMemoryUsage(isolateRef.getId(), new GetMemoryUsageConsumer() {
-        private int errorCount = 0;
-
         @Override
         public void received(MemoryUsage usage) {
           memoryUsage.add(usage);
-
-          // Only update when we're done collecting from all isolates.
-          if ((memoryUsage.size() + errorCount) == isolateCount) {
-            notifyListeners(memoryUsage);
-          }
+          latch.countDown();
         }
 
         @Override
         public void received(Sentinel sentinel) {
-          errorCount++;
+          latch.countDown();
         }
 
         @Override
         public void onError(RPCError error) {
-          errorCount++;
+          // {"jsonrpc":"2.0","error":{"code":-32603,"message":"UnimplementedError..."}}
+          if (error.getCode() == -32603) {
+            handleMemoryApiNotSupported();
```

Comment:
Can you just check the vm service version since we should know which version started supporting this API? This is what we do for backwards compatibility in DevTools for recently added vm service APIs.

Reply:
The problem is that dwds will return unimplemented codes for a lot of apis even though its vm service version has been updated. That really just means we will accept a request for that method but we might return an unimplemented response still.

Reply:
Gotcha that makes sense. Nevermind then 👍